source/eventcore/drivers/posix/sockets.d

```diff
@@ -117,7 +117,8 @@ final class PosixEventDriverSockets(Loop : PosixEventLoop) : EventDriverSockets
 	{
 		assert(on_connect !is null);
 
-		auto sockfd = createSocket(address.addressFamily, SOCK_STREAM);
+		// @trusted to escape DIP1000's `scope` check
+		auto sockfd = () @trusted { return createSocket(address.addressFamily, SOCK_STREAM); }();
```

Comment:
Just curious, is this necessary because `Address.addressFamily()` is missing a `scope` qualifier? (could make sense to move the `@trusted` inside the parameter list then)

Reply:
The `addressFamily` method in Phobos would need to be changed for this not to be necessary.
The error itself is (different location but same stuff):
```
source/eventcore/drivers/posix/sockets.d(708,30): Error: scope variable bind_address assigned to non-scope parameter this calling std.socket.Address.addressFamily
```